src/volttron/server/aip.py

```diff
@@ -483,56 +484,74 @@ def land_agent(self, agent_wheel):
             raise
         return agent_uuid
 
+    def backup_agent_data(self, agent_uuid, publickey, secretkey, vip_identity):
+        backup_agent_file = None
+        if agent_uuid:
+            _log.debug(f"There is an existing agent {agent_uuid}")
+            old_agent_data_dir = self.get_agent_data_dir(agent_uuid)
+            if os.listdir(old_agent_data_dir):
+                # And there is data to backup
+                backup_agent_file = "/tmp/{}.tar.gz".format(agent_uuid)
+                with tarfile.open(backup_agent_file, "w:gz") as tar:
+                    tar.add(old_agent_data_dir, arcname=os.path.sep)  # os.path.basename(source_dir))
+            # current agent's keystore overrides any keystore data passed
+            keystore = self.__get_agent_keystore__(vip_identity, publickey, secretkey)
+            publickey = keystore.public
+            secretkey = keystore.secret
+
+            _log.info('Removing previous version of agent "{}"\n'.format(vip_identity))
```

Comment:
Nit. Instead of `format`, f-strings would work here.

Reply:
fixed this and a few other places but this type of formatting is still around in lots of places in our code and I assume would get replaced slowly as we other changes in the module

Reply:
Reverted the changes done for string formatting as there were lots of these and it makes looking at other functional changes difficult. Will create separate PR for string format changes